### DISCUSSION TITLE
Support Import of SVG and Use as a Component

### DISCUSSION THREAD

**Discussion by justinnoel** [withastro/roadmap#38] [Proposal]
RE: https://discord.com/channels/830184174198718474/915244207705292821/915580508895842347

The [Imports](https://docs.astro.build/guides/imports/#other-assets) docs, indicate that SVG support is allowed. The docs suggest using the import as the source for an image.

```astro
---
import svgReference from './image.svg';
---
<img src={imgReference} />;
```

However, this is not ideal as actual SVG elements are more accessible than images.

I'm proposing supporting the import and usage of SVGs as components in Astro files like:

```astro
---
import Logo from './logo.svg';
---
<h1>Awesome App!</h1>
<Logo />
```

In the interim, Muhymin has offered this solution to the problem:

```astro
---
import Logo from './logo.svg?raw';
---
<h1>Awesome App!</h1>
<Logo />
```

**Comment by FredKSchott** [comment]
I really like this idea! I'd love to support this

That workaround is fine for now, but the problem in the future is that we are preparing to add XSS scripting protection, which means that injecting a raw, unescaped HTML like that would fail as its written. Instead, you would need something like this RFC: 

```astro
---
// RFC: https://github.com/withastro/astro/issues/1827
import logoHtml from './logo.svg?raw';
---
<h1>Awesome App!</h1>
<div set:innerhtml={logoHtml} />
```

If you wanted to get rid of `?raw` and import an Astro component by default, we'd also need some way to make this work across React/Vue/Svelte components. They may also want to import an SVG, in which case it shouldn't always be a React component. Maybe the easiest thing would be to default SVGs to always import in `raw` mode, by default.

**Comment by jonathantneal** [comment]
In theory, I think we can trial **renderers** for this (and images) in user-land, since your request seems to focus exclusively on Astro.

**Comment by justinnoel** [comment]
I like the RFFC 1827 idea.

Maybe something like (or exactly like) @natemoo-re's new `astro-icon` is the solution I'm looking for. https://github.com/natemoo-re/astro-icon.

As Jonathan indicated, maybe it's not a problem Astro itself should be solving and would just get too complicated with other libraries.

So, let's consider this "closed" and I'll try out all the ideas.

Thanks for the feedback.

**Comment by cxa** [comment]
Before https://github.com/withastro/roadmap/issues/699 is out, you can try https://github.com/cxa/astro-inline-svg:

```astro
---
import Svg from "@cxa/astro-inline-svg";
import sunRiseSVG from "lucide-static/icons/sunrise.svg?raw";
---

<Svg raw={sunRiseSVG} class="sun-rise" stroke-width={1.5} />
```

**Comment by florian-lefebvre** [comment]
Astro supports this since a few majors, closing
